### PR TITLE
hotfix: fix #635

### LINKED ISSUE #635: Swagger and custom POST operation
` /**
 * @ApiResource(
 *     collectionOperations={
 *         "create"={
 *            "path"="/customers/create",
 *            "method"="POST",
 *            "controller"="AppBundle:Customer:create",
 *            "hydra_context"={"@type"="hydra:Operation", "hydra:title"="Création de client", "returns"="jsonld"}
 *         }
 *     },`

this creates two operations in doc (POST /customers and POST customer/create)

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #635 
| License       | MIT
| Doc PR        | 

@napalm33700 Could you please test this and tell me if its work ?

### CODE REVIEW

File: src/Swagger/DocumentationNormalizer.php [hunk 1]
```diff
@@ -163,7 +163,7 @@ public function normalize($object, $format = null, array $context = [])
                         if (null !== $path) {
                             $method = $this->operationMethodResolver->getCollectionOperationMethod($resourceClass, $customOperation);
                             $customSwaggerOperation = $this->getSwaggerOperation($resourceClass, $resourceMetadata, $customOperation, [$method], $prefixedShortName, true, $definitions, $method, $object->getMimeTypes());
-
+                            unset($itemOperationsDocs[$resourceClassIri]);
```
Comment (dunglas) [inline src/Swagger/DocumentationNormalizer.php]: The code can be made more readable (and more performant) by testing first of a custom operation applies and defaulting to the global one only if no specific is set.

What do you think @Simperfit?

Btw, can you add a non-regression test?

Comment (Simperfit) [inline src/Swagger/DocumentationNormalizer.php]: I totally agree, I made this WIP for this purpose to improve this part of the code

I will add one.

**Comment by Simperfit** [comment]
Comments addressed @dunglas

**Comment by napalm33700** [comment]
Sorry I was busy...  
With fix #635 no more unwanted operation but now I have duplicates (annotation is for POST only)
api

**Comment by Simperfit** [comment]
@napalm33700 What do you mean by dupplicated ? is the get unwanted ?

**Comment by napalm33700** [comment]
It is not really duplicates but unwanted operations 

> * @ApiResource(
 *     attributes={"filters"={"article.search"}},
 *     collectionOperations={
 *         "get"={"method"="GET", "normalization_context"={"groups"={"article"}}},
 *         "stock"={
 *            "path"="/articles/stock",
 *            "method"="POST",
 *            "controller"="AppBundle:Article:stock",
 *            "hydra_context"={"@type"="hydra:Operation", "hydra:title"="Stock Articles", "returns"="jsonld"}
 *         }
 *     },
 *     itemOperations={
 *         "get"={"method"="GET", "normalization_context"={"groups"={"article"}}},
 *     }
 * )

It should be  
GET /articles
POST / articles/stock
GET /articles/{id}

**Comment by Ener-Getick** [comment]
Oops I also opened https://github.com/api-platform/core/pull/648... but from what I see your PR introduces another bug by always overriding `$resourceClassIri`.

**Comment by Simperfit** [comment]
@napalm33700 Could you please test with @Ener-Getick's PR, if it's works for you then ill close this one ;).

@Ener-Getick  You are right, I feels that the way you refactored this part of the code is better.

**Comment by dunglas** [comment]
@Simperfit can we close this one?

**Pull request closed without merging**